packages/adblocker/src/resources.ts

```diff
@@ -297,8 +297,18 @@ export default class Resources {
     }
   }
 
-  public getResource(name: string): { body: string; contentType: string; dataUrl: string } {
-    const { body, contentType } = this.resourcesByName.get(name) || getResourceForMime(name);
+  public getResource(name: string): {
+    name: string;
+    body: string;
+    contentType: string;
+    dataUrl: string;
+  } {
+    const resource = this.resourcesByName.get(name) || getResourceForMime(name);
+    const { contentType, body } = resource;
+
+    if ('name' in resource) {
+      name = (resource as Resource).name;
```

Comment:
please do not reassign the function argument - why the linter does not complain about it ?

Reply:
https://eslint.org/docs/latest/rules/no-param-reassign

Reply:
`no-param-reassign` is not enabled currently. `recommended-typechecked` from `typescript-eslint` doesn't pick up `no-param-reassign`.

Reply:
refs https://github.com/ghostery/adblocker/issues/4883